### PR TITLE
[14.0][ADD] product_configurator_restriction_policy

### PR DESCRIPTION
**Pull request by patrickrwilson**
This adds an optional way to process the restrictions.

### CODE REVIEW

**Comment by patrickrwilson** [comment]
@dreispt pre-commit is complaining that the method is too complex. I've simplified it but still doesn't like it. I'm not sure why, ideas?

product_configurator_restriction_policy/models/product_config.py:11:5: C901 'ProductConfigSession.values_available' is too complex (17)

**Comment by dreispt** [comment]
The strategy I would use is to break parts of the big function into smaller helper functions.

**Comment by dreispt** [comment]
Usually, when you feel the urge to add a code comment, that's often a code smell indication that you could extract the following code into its own function.

File: product_configurator_restriction_policy/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    restriction_policy = fields.Selection(
+        [("standard", "Standard"), ("sequential", "Sequential")],
+        string="Restriction Policy",
+        default="standard",
+        help="""Standard: When the system is calculating the available values from
```
Comment (bizzappdev) [inline product_configurator_restriction_policy/models/res_config_settings.py:14]: why not use "config_parameter" 
something like https://github.com/odoo/odoo/blob/14.0/addons/account/models/res_config_settings.py#L66

**Review by bizzappdev — CHANGES_REQUESTED** [review]

File: product_configurator_restriction_policy/models/res_config_settings.py (hunk 1 above)
Comment (patrickrwilson) [reply product_configurator_restriction_policy/models/res_config_settings.py:14]: I didn't know this existed, thanks! that is way easier. I'll implement that soon.

**Comment by patrickrwilson** [comment]
@bizzappdev @dreispt ready for your review. Thank you both for your help.

**Comment by bizzappdev** [comment]
Code LGTM. some love to test cases will be really great.

**Comment by patrickrwilson** [comment]
> Code LGTM. some love to test cases will be really great.

Yeah, we need to work on test cases for all of these, including the core ones which is coming later.

File: product_configurator_restriction_policy/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "website": "https://github.com/OCA/product-configurator",
+    "depends": ["product_configurator"],
+    "data": [
+        "views/res_config_settings_view.xml",
+        "views/product_view.xml",
+    ],
+    "development_status": "Beta",
+    "installable": True,
+    "application": True,
+    "auto_install": False,
```
Comment (dreispt) [inline product_configurator_restriction_policy/__manifest__.py]:
```suggestion
    "application": False,
```

**Comment by dreispt** [comment]
Ping @patrickrwilson

**Pull request merged**